platform/o.n.bootstrap/src/org/netbeans/JarClassLoader.java

```diff
@@ -893,6 +919,30 @@ protected URL doGetResource(String name) throws MalformedURLException {
                 throw new IOException(ex);
             }
         }
+
+        protected byte[] readClass(String path) throws IOException {
+            /* "FileOutputStream and FileInputStream complicate GC because they both override
+            'finalize()'. Switching to NIO API creates Stream without finalizers and thus relieving
+            GC." However, using NIO means IO operations can fail with a ClosedByInterruptException,
+            for instance if client code starts a new thread which needs to load some new classes,
+            and then interrupts the thread before class loading has finished. See NETBEANS-1197.
+            Note that the old java.io.InterruptedIOException, which could in theory be called even
+            from FileOutputStream/FileInputStream, was seldom seen in practice; see JDK-4385444. */
+            try {
+                return readClassHelper(path, true);
+            } catch (ClosedByInterruptException ex) {
+                LOGGER.log(Level.INFO,
```

Comment:
I would log the event with lower `Level`; the interruption can interfere with class loading at any time for valid reason; no need to alert the user / administrator by logging at INFO.

Reply:
I'd be inclined to keep it at INFO, to help in the following scenario: A user encounters some intermittent concurrency bug involving the classloader system, and submits a console log. A developer now reads through the JarClassLoader source, encounters the ClosedByInterruptException-related code, and thinks--"gee, I wonder if this code is somehow related to the bug I'm investigating". If the log message is left at INFO, and the console log does not contain the message, the developer knows they can safely ignore the ClosedByInterruptException-related code, and that the bug is somewhere else. If we don't print the log message by default, the developer may spend lots of time trying to exercise this odd code path to see if it's related to whatever bug they are investigating.

The event should be rare, and when it does happen, it would be unlikely to happen more than once in a given JVM session (since the class does get loaded in the end). Though if you want, I could reword the log message to emphasize the fact that it's really not an error.